Multiple instances of relying on (data-dependent) subdirectories being precreated

Since the aggregator sorts data to node-separated subdirectories, it would be nice if it did not rely on all those being precreated.

In short, this typically means using os.makedirs rather than os.mkdir.

I have ran in to the same issue. Configuring the aggregator and running it gives:

```
[root@hebbe-c1 local]# sams-aggregator.py 
2020-01-14 09:15:23,346 __main__:ERROR Failed to do aggregation
2020-01-14 09:15:23,346 __main__:ERROR Failed to mkdir '/local/sams_test/archive/10000' 
Traceback (most recent call last):
  File "/usr/bin/sams-aggregator.py", line 88, in start
    l.commit()
  File "/usr/lib/python2.7/site-packages/sams/loader/File.py", line 82, in commit
    assert False, "Failed to mkdir '%s' " % out_path
AssertionError: Failed to mkdir '/local/sams_test/archive/10000' 
Traceback (most recent call last):
  File "/usr/bin/sams-aggregator.py", line 104, in <module>
    Main().start()
  File "/usr/bin/sams-aggregator.py", line 96, in start
    l.error()
  File "/usr/lib/python2.7/site-packages/sams/loader/File.py", line 62, in error
    assert False, "Failed to mkdir '%s' " % out_path
AssertionError: Failed to mkdir '/local/sams_test/error/10000' 
```

`os.makedirs` would be very convenient.

Best, Hugo

--- a/sams-post-receiver.py
+++ b/sams-post-receiver.py
@@ -36,11 +36,11 @@ class Receiver(MethodView):
 
             if not os.path.isdir(base_path):
                 try:
-                    os.mkdir(base_path)
+                    os.makedirs(base_path)
                 except Exception as err:
                     # Handle possible raise from other process
                     if not os.path.isdir(base_path):
-                        assert False, "Failed to mkdir '%s' " % base_path
+                        assert False, "Failed to makedirs '%s' " % base_path
 
         tfilename = ".%s" % filename
         try:

--- a/sams/loader/File.py
+++ b/sams/loader/File.py
@@ -55,11 +55,11 @@ class Loader(sams.base.Loader):
         out_path = os.path.join(self.error_path,self.current_file['path'])
         if not os.path.isdir(out_path):
             try:
-                os.mkdir(out_path)
+                os.makedirs(out_path)
             except Exception as err:
                 # Handle possible raise from other process
                 if not os.path.isdir(out_path):
-                    assert False, "Failed to mkdir '%s' " % out_path
+                    assert False, "Failed to makedirs '%s' " % out_path
 
         # Rename file to error directory
         os.rename(os.path.join(self.in_path,self.current_file['path'],self.current_file['file']),
@@ -75,11 +75,11 @@ class Loader(sams.base.Loader):
         out_path = os.path.join(self.archive_path,self.current_file['path'])
         if not os.path.isdir(out_path):
             try:
-                os.mkdir(out_path)
+                os.makedirs(out_path)
             except Exception as err:
                 # Handle possible raise from other process
                 if not os.path.isdir(out_path):
-                    assert False, "Failed to mkdir '%s' " % out_path
+                    assert False, "Failed to makedirs '%s' " % out_path
 
         # Rename file to archive directory
         os.rename(os.path.join(self.in_path,self.current_file['path'],self.current_file['file']),

--- a/sams/output/File.py
+++ b/sams/output/File.py
@@ -63,11 +63,11 @@ class Output(sams.base.Output):
 
             if not os.path.isdir(base_path):
                 try:
-                    os.mkdir(base_path)
+                    os.makedirs(base_path)
                 except Exception as err:
                     # Handle possible raise from other process
                     if not os.path.isdir(base_path):
-                        assert False, "Failed to mkdir '%s' " % base_path
+                        assert False, "Failed to makedirs '%s' " % base_path
 
         try:
             with open(os.path.join(base_path,tfilename),"w") as file:

